### DISCUSSION TITLE
[Pagination]Please remove the limit on the number of max pages

### DISCUSSION THREAD

**Discussion by seasoncool** [grafana/grafana-infinity-datasource#704] [General]
Dear all,

Our system's API allows only one parameter to be passed for each query, and unfortunately, we cannot modify the logic of this backend API to enable batch queries.

We were excited to discover the pagination feature, but it only allows us to retrieve information for a maximum of 5 pages at a time.

Is it possible to lift the restriction on the maximum number of pages here?
 We have a large number of parameters to pass one by one, and the current limitation of 5 pages restricts the number of our requests.

or any ideas are welcomed.

![image](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/6467295/cb318928-0fbd-4cda-85b1-26dafbf44fa5)

**Comment by yesoreyeram** [comment]
You mentioned

> unfortunately, we cannot modify the logic of this backend API to enable batch queries

The restriction of 5 is for similar reason. We don't want to put accidental stress on any system. That may cause DenialOfService.

**Reply by seasoncool** [reply]
Is there another way to achieve this requirement using infinity ？ 
any ideas are welcomed...
